src/sentry/sentry_metrics/querying/api.py

```diff
@@ -672,13 +932,20 @@ def run_metrics_query(
 
     # Parsing the input and iterating over each timeseries.
     parser = QueryParser(fields=fields, query=query, group_bys=group_bys)
-    for timeseries in parser.generate_queries(environments=environments):
-        query = (
-            base_query.set_query(timeseries)
-            .set_rollup(Rollup(interval=interval))
-            .set_limit(SNUBA_QUERY_LIMIT)
+    for field, timeseries in parser.generate_queries(environments=environments):
+        query = base_query.set_query(timeseries).set_rollup(Rollup(interval=interval))
+
+        # We will apply the order by if it only matches the field. This is done since for now we don't support a custom
+        # since for order bys.
+        query_order_by = None
+        if order_by and field == order_by.removeprefix("-"):
+            query_order_by = order_by
```

Comment:
This is not very nice, but it's the simplest solution especially since we are not sure where to head w.r.t. the API with multiple queries.